docs/modules/SDDs/pages/0025-commodore-component-instantiation.adoc

```diff
@@ -0,0 +1,100 @@
+= SDD 0025 - Commodore Component Instantiation
+
+:sdd_author:    Simon Gerber
+:sdd_owner:     Simon Gerber
+:sdd_reviewers: Reviewer names
+:sdd_date:      2020-10-23
+:sdd_status:    draft
+include::partial$meta-info-table.adoc[]
+
+[NOTE]
+.Summary
+====
+This SDD describes possible options to allow Commodore components to be instantiated multiple times per cluster.
+It additionally defines the option we've chosen to implement.
+====
+
+== Motivation
+
+Currently Commodore components don't support installing multiple instances of the software they manage.
+There are already a some components for which installing multiple instances is a common use case, such as the https://github.com/projectsyn/component-nfs-client-provisioner/[nfs-client-provisioner component].
+
+=== Goals
+
+* Define how we want to allow Commodore components to be instantiated multiple times per cluster.
+
+=== Non-Goals
+
+* Define a solution for managing multiple instances of services for the customer on a Syn-enabled cluster.
+
+== Design Proposal
+
+Commodore will provide a well-defined structure for instantiating components and providing per-instance parameters.
+Commodore will provide the instance's identifier ("name") to the component in a well-defined way.
+Component authors must explicitly mark their components as supporting instantiation.
```

Comment:
That is a strange sentence.

```suggestion
Component authors must explicitly declare that their component is ready for being instantiated multiple times.
```

Well, not that happy with that wording either. Feel free to come up with something better.

Reply:
I took half of your suggestion and half of the original :)